### ISSUE TITLE
Car route should not begin/end on a motorway

### ISSUE THREAD

**Issue by polyscias** [abrensch/brouter#682]
In most well developed countries it is forbidden to stop on a motorway (highway=motorway/trunk), so when a router is searching for the begin/end of the route is should exclude motorways. Example of things going wrong: 

http://brouter.de/brouter-web/#map=16/51.9858/7.5655/standard&lonlats=7.555976,51.984186;7.550569,51.985726&profile=car-eco

**Comment by poutnikl** [comment]
But the route start/stop points do not necessarily mean you physically start/stop there.

**Comment by polyscias** [comment]
> But the route start/stop points do not necessarily mean you physically start/stop there.

Indeed, drivers should never 100% trust the routes that are calculated and keep thinking themselves. Still it would be good to have the route as good as possible.

> It may mean you already are on a motorway (common case if you are a "navigator" during abroad car trip) or you just want to know how to get on the motorway.

I do not know any car routing device that is integrating brouter and a driver should not operate a smartphone during driving so the only remaining case is where a co-driver uses his/her smartphone.

This argument only holds for the "From", for the "To" I see not a single good reason it should end on a motorway.

Implementation should be relatively simple, find these points is separate functionality and you just have to filter out highway=motorway/trunk running the algorithm.

**Comment by devemux86** [comment]
> It may mean you already are on a motorway (common case if you are a "navigator" during abroad car trip)

Correctly. :+1:

Routing (automatic or not) is necessary for navigation.
Users expect that routes can start anywhere (including motorways).

Routing should not start arbitrarily on random roads other than the road the user is on.

---

[Cruiser](https://github.com/devemux86/cruiser) is a navigator using BRouter for car and motorcycle driving.

**Comment by poutnikl** [comment]
> Indeed, drivers should never 100% trust the routes that are calculated and keep thinking themselves. Still it would be good to have the route as good as possible.

My post was not in context of routing reliability, but about starting/stopping the routing on motorways intentionally.

> 
> I do not know any car routing device that is integrating brouter and a driver should not operate a smartphone during driving so the only remaining case is where a co-driver uses his/her smartphone.

You have probably missed the navigator. I was not speaking about a driver on a motorway. I occasionally perform navigator role on abroad vacation trips with friends and start routing at motorway in real time frequently.

> This argument only holds for the "From", for the "To" I see not a single good reason it should end on a motorway.

The task to get on a motorway is very frequent, as it may be challenging to orientation. Getting already on the motorway, you often do not need navigation for hours.

**Comment by polyscias** [comment]
> You have probably missed the navigator. 

Yes, I initially missed that but already got it in my previous reply and proposed to limit it to the destination point.

> it looks like you want a kind of safety net again random manual choosing of motorway

Not directly, I want the router to give the best route which a human should expect.

@devemux86, @poutnikl do you think [this route](http://brouter.de/brouter-web/#map=16/51.9850/7.5573/standard&lonlats=7.550569,51.985726;7.555976,51.984186&profile=car-eco) for somebody that visited Buschkamp and want to driver home should end on the highway?

**Comment by poutnikl** [comment]
What is the point of not selecting of destination properly or of not having home as the saved favourite of the house approaching road?

**Comment by devemux86** [comment]
> They is no reason not to allow motorway as a start/destination, if a car is allowed to be going there

Indeed, any routing (driving, cycling, walking) should work everywhere that is allowed.

Or perhaps mean the snap of waypoints to nearby roads?
This is different and happens before routing. There is a parameter of waypoint catching range.

**Comment by devemux86** [comment]
> Or perhaps mean the snap of waypoints to nearby roads?

For example [GraphHopper API](https://docs.graphhopper.com/#operation/getRoute!in=query&path=snap_prevention&t=request) has the parameter "snap_prevention":

_"Optional parameter to avoid snapping to a certain road class or road environment. Currently supported values are motorway, trunk, ferry, tunnel, bridge and ford."_
